analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.ExceptionCandidate.cs

```diff
@@ -572,5 +573,75 @@ End Sub
             validator.ValidateTagOrder("BeforeTry", "InTry");
             validator.ExitStates.Should().HaveCount(0);
         }
+
+#if NET
+        [TestMethod]
+        public void ExceptionCandidate_ImplicitIndexerReference()
+        {
+            const string code = @"
+using System;
+
+class Sample
+{
+    public int this[Index i] => 0;
+
+    public void Method()
+    {
+        var tag = ""BeforeTry"";
+        try
+        {
+            tag = ""InTry"";
+            _ = this[^0];
+        }
+        catch
+        {
+            tag = ""UnreachableInCatch"";
+        }
+        tag = ""AfterCatch"";
+    }
+}";
+
+            var validator = CreateCSharp10Validator(code);
+            // IImplicitIndexerReferenceOperation is not generated
+            validator.ValidateTagOrder("BeforeTry", "InTry", "AfterCatch");
+            validator.ExitStates.Should().HaveCount(1).And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_Range()
+        {
+            const string code = @"
+using System;
+
+class Sample
+{
+    public int this[Range r] => 0;
```

Comment:
We can do (almost) the same thing with `", int[] array"` additional argument. To remove the extra scaffolding.

Reply:
It looks like they are handled differently. Actually, we will need both tests.